### PR TITLE
Improvement: design

### PR DESCRIPTION
**Pull request by Zorin95670**

### CODE REVIEW

File: src/components/dialog/ImportProjectDialog.vue [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import DefaultDialog from 'src/components/dialog/DefaultDialog';
+import ImportProjectForm from 'src/components/form/ImportProjectForm';
+import { useRouter } from 'vue-router';
+
+const router = useRouter();
+
+function importProject(projectId) {
+  DialogEvent.next({ type: 'close', key: 'ImportProject' });
+  return router.push(`/modelizer/${projectId}/model`);
+}
```
Comment (justinehell) [inline src/components/dialog/ImportProjectDialog.vue:26]: Missing doc

File: src/components/form/ImportProjectForm.vue [hunk 2]
```diff
@@ -0,0 +1,103 @@
+<template>
+  <q-form
+    @submit="onSubmit"
+    data-cy="import-project-form"
+    class="q-gutter-md import-project-form">
```
Comment (justinehell) [inline src/components/form/ImportProjectForm.vue:5]: jump line before `>`

File: src/pages/HomePage.vue [hunk 3]
```diff
@@ -6,39 +6,24 @@
...
-        @click="createProject"
-      />
       <div class="fit row justify-center">
         <ProjectGrid
           class="col-md-8"
           :projects="getProjects()"
         />
       </div>
     </div>
+    <import-project-dialog></import-project-dialog>
```
Comment (justinehell) [inline src/pages/HomePage.vue:19]: `<import-project-dialog/>`

File: tests/unit/components/card/ProjectCard.spec.js [hunk 4]
```diff
@@ -25,4 +25,28 @@ describe('Test component: ProjectCard', () => {
       });
     });
   });
+
+  describe('Test functions', () => {
+    describe('Test function: getProjectImage', () => {
+      it('should return correct image name', async () => {
```
Comment (justinehell) [inline tests/unit/components/card/ProjectCard.spec.js:31]: `correct` -> `corresponding` ?

File: src/components/grid/ProjectGrid.vue [hunk 5]
```diff
@@ -1,38 +1,70 @@
...
+        <q-btn
+            outline
+            no-caps
+            class="q-ml-xl"
+            color="primary"
+            icon="fa-solid fa-plus"
+            data-cy="new-project"
+            :label="$t('actions.home.newProject')"
+            @click="createProject"
+        />
```
Comment (justinehell) [inline src/components/grid/ProjectGrid.vue:15]: wrong indent

File: src/components/grid/ProjectGrid.vue [hunk 6]
```diff
@@ -1,38 +1,70 @@
...
+        <q-btn
+            outline
+            no-caps
+            class="q-ml-sm"
+            color="primary"
+            icon="fa-solid fa-cloud-arrow-down"
+            data-cy="import-project"
+            :label="$t('actions.home.importProject')"
+            @click="DialogEvent.next({ type: 'open', key: 'ImportProject' })"
+        />
```
Comment (justinehell) [inline src/components/grid/ProjectGrid.vue:25]: wrong indent

File: src/composables/Project.js [hunk 7]
```diff
@@ -88,6 +88,27 @@ export function initProject(project) {
   return git.init({ fs, dir: `/${project.id}` });
 }
 
+/**
+ * Save and clone project from git in local storage.
+ * @param {Project} project - Project to save
```
Comment (justinehell) [inline src/composables/Project.js:93]: dot at the end

File: tests/unit/components/grid/ProjectGrid.spec.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    });
+  });
+
+  describe('Test functions', () => {
+    describe('Test function: createProject', () => {
+      it('should create, init and redirect project ', async () => {
+        await wrapper.vm.createProject();
+        expect(createProjectTemplate).toBeCalled();
+        expect(initProject).toBeCalled();
+        expect(push).toBeCalled();
```
Comment (justinehell) [inline tests/unit/components/grid/ProjectGrid.spec.js:57]: maybe check `expect(push).toBeCalledWith()` ?

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/components/dialog/ImportProjectDialog.vue (hunk 1 above)
Comment (Zorin95670) [reply src/components/dialog/ImportProjectDialog.vue:26]: done

File: src/components/form/ImportProjectForm.vue (hunk 2 above)
Comment (Zorin95670) [reply src/components/form/ImportProjectForm.vue:5]: done

File: src/pages/HomePage.vue (hunk 3 above)
Comment (Zorin95670) [reply src/pages/HomePage.vue:19]: done

File: src/components/grid/ProjectGrid.vue (hunk 5 above)
Comment (Zorin95670) [reply src/components/grid/ProjectGrid.vue:15]: done

File: src/components/grid/ProjectGrid.vue (hunk 6 above)
Comment (Zorin95670) [reply src/components/grid/ProjectGrid.vue:25]: done

File: src/components/grid/ProjectGrid.vue [hunk 9]
```diff
@@ -1,38 +1,70 @@
...
+      </div>
+      <div class="row">
         <ProjectCard
-          class="q-pa-xs col-xs-12 col-sm-4 col-md-3"
-          :project="props.row"
+          class="q-ma-sm"
+          v-for="project in projects"
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/components/grid/ProjectGrid.vue:30]: done

File: src/composables/Project.js (hunk 7 above)
Comment (Zorin95670) [reply src/composables/Project.js:93]: done

File: src/components/form/ImportProjectForm.vue [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+    <q-input
+      filled
+      v-model="repository"
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/components/form/ImportProjectForm.vue:8]: done

File: tests/unit/components/grid/ProjectGrid.spec.js (hunk 8 above)
Comment (Zorin95670) [reply tests/unit/components/grid/ProjectGrid.spec.js:57]: done

**Review by k-y0u — APPROVED** [review]

**Review by justinehell — APPROVED** [review]

**Pull request merged**